clang/unittests/Driver/DXCModeTest.cpp

```diff
@@ -68,25 +68,25 @@ TEST(DxcModeTest, TargetProfileValidation) {
   IntrusiveRefCntPtr<DiagnosticOptions> DiagOpts = new DiagnosticOptions();
   DiagnosticsEngine Diags(DiagID, &*DiagOpts, DiagConsumer);
 
-  validateTargetProfile("-Tvs_6_0", "dxil--shadermodel6.0-vertex",
+  validateTargetProfile("-Tvs_6_0", "dxilv1.0--shadermodel6.0-vertex",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Ths_6_1", "dxil--shadermodel6.1-hull",
+  validateTargetProfile("-Ths_6_1", "dxilv1.1--shadermodel6.1-hull",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tds_6_2", "dxil--shadermodel6.2-domain",
+  validateTargetProfile("-Tds_6_2", "dxilv1.2--shadermodel6.2-domain",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tds_6_2", "dxil--shadermodel6.2-domain",
+  validateTargetProfile("-Tds_6_2", "dxilv1.2--shadermodel6.2-domain",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tgs_6_3", "dxil--shadermodel6.3-geometry",
+  validateTargetProfile("-Tgs_6_3", "dxilv1.3--shadermodel6.3-geometry",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tps_6_4", "dxil--shadermodel6.4-pixel",
+  validateTargetProfile("-Tps_6_4", "dxilv1.4--shadermodel6.4-pixel",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tcs_6_5", "dxil--shadermodel6.5-compute",
+  validateTargetProfile("-Tcs_6_5", "dxilv1.5--shadermodel6.5-compute",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tms_6_6", "dxil--shadermodel6.6-mesh",
+  validateTargetProfile("-Tms_6_6", "dxilv1.6--shadermodel6.6-mesh",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tas_6_7", "dxil--shadermodel6.7-amplification",
+  validateTargetProfile("-Tas_6_7", "dxilv1.7--shadermodel6.7-amplification",
                         InMemoryFileSystem, Diags);
-  validateTargetProfile("-Tlib_6_x", "dxil--shadermodel6.15-library",
+  validateTargetProfile("-Tlib_6_8", "dxilv1.8--shadermodel6.8-library",
```

Comment:
There is, in fact, a lib_6_x target that maps to a more unconstrained offline linking target (with minor version = 15) which is not expected to be compatible between compiler versions.  Do we care to model that in clang at this time?  This change removes this target, so I just wanted to know if that was a deliberate decision.

Reply:
> There is, in fact, a lib_6_x target that maps to a more unconstrained offline linking target (with minor version = 15) which is not expected to be compatible between compiler versions. Do we care to model that in clang at this time? This change removes this target, so I just wanted to know if that was a deliberate decision.

What is the expected DXIL version of DXIL Ops generated using the target profile `lib_6_x`?

Reply:
`lib_6_x` should basically map to "whatever the latest DXIL version is"

Reply:
> `lib_6_x` should basically map to "whatever the latest DXIL version is"

In that case should the call be changed to 
```
validateTargetProfile("-Tlib_6_x", "dxilv1.8--shadermodel6.8-library",
                        InMemoryFileSystem, Diags);
```

and the option `-T<kind>6_x` be considered equivalent to `-T<kind>6_8` (as of now)?

Reply:
I think the change you proposed is correct (`-Tlib_6_x` maps to `dxilv1.8--shadermodel6.15-library`), but that is crucially _not_ making it equivalent to `-Tlib_6_8`.

Today when DXC compiles the `lib_6_8` target the resulting binary encodes the shader model as 6.8 and DXIL version as 1.8.

Today when DXC compiles the `lib_6_x` target the resulting binary encodes the shader model as 6.15 and DXIL version as 1.8.

Which means:
* `lib_6_8` -> `dxilv1.8--shadermodel6.8-library`
* `lib_6_x` -> `dxilv1.8--shadermodel6.15-library`

Reply:
> `lib_6_x` should basically map to "whatever the latest DXIL version is"

Pushed a change to consider `lib_6_x` as a valid profile equivalent to `lib_6_8`.